src/controllers/scripting/javascriptplayerproxy.cpp

```diff
@@ -0,0 +1,125 @@
+    connect(m_pTrackPlayer,
+            &BaseTrackPlayer::newTrackLoaded,
+            this,
+            &JavascriptPlayerProxy::slotTrackLoaded);
+    connect(m_pTrackPlayer,
+            &BaseTrackPlayer::playerEmpty,
+            this,
+            &JavascriptPlayerProxy::trackUnloaded);
+    connect(this,
+            &JavascriptPlayerProxy::trackChanged,
+            this,
+            &JavascriptPlayerProxy::slotTrackChanged);
+    if (m_pTrackPlayer && m_pTrackPlayer->getLoadedTrack()) {
+        slotTrackLoaded(pTrackPlayer->getLoadedTrack());
+    }
+}
+
+void JavascriptPlayerProxy::slotTrackLoaded(TrackPointer pTrack) {
+    m_pCurrentTrack = pTrack;
+    if (pTrack != nullptr) {
+        connect(pTrack.get(),
+                &Track::artistChanged,
+                this,
+                &JavascriptPlayerProxy::artistChanged);
+        connect(pTrack.get(),
+                &Track::titleChanged,
+                this,
+                &JavascriptPlayerProxy::titleChanged);
+        connect(pTrack.get(),
+                &Track::albumChanged,
+                this,
+                &JavascriptPlayerProxy::albumChanged);
+        connect(pTrack.get(),
+                &Track::albumArtistChanged,
+                this,
+                &JavascriptPlayerProxy::albumArtistChanged);
+        connect(pTrack.get(),
+                &Track::genreChanged,
+                this,
+                &JavascriptPlayerProxy::genreChanged);
+        connect(pTrack.get(),
+                &Track::composerChanged,
+                this,
+                &JavascriptPlayerProxy::composerChanged);
+        connect(pTrack.get(),
+                &Track::groupingChanged,
+                this,
+                &JavascriptPlayerProxy::groupingChanged);
+        connect(pTrack.get(),
+                &Track::yearChanged,
+                this,
+                &JavascriptPlayerProxy::yearChanged);
+        connect(pTrack.get(),
+                &Track::trackNumberChanged,
+                this,
+                &JavascriptPlayerProxy::trackNumberChanged);
+        connect(pTrack.get(),
+                &Track::trackTotalChanged,
+                this,
+                &JavascriptPlayerProxy::trackTotalChanged);
+    }
+    emit trackChanged();
+    emit trackLoaded();
+}
+
+void JavascriptPlayerProxy::slotLoadingTrack(TrackPointer pNewTrack, TrackPointer pOldTrack) {
+    VERIFY_OR_DEBUG_ASSERT(pOldTrack == m_pCurrentTrack) {
+        qWarning() << "QML Player proxy was expected to contain "
+                   << pOldTrack.get() << "as active track but got"
+                   << m_pCurrentTrack.get();
+    }
+
+    if (pNewTrack.get() == m_pCurrentTrack.get()) {
+        emit trackLoading();
+        return;
+    }
+
+    const TrackPointer pTrack = m_pCurrentTrack;
+    if (pTrack != nullptr) {
+        disconnect(pTrack.get(), nullptr, this, nullptr);
+    }
+    m_pCurrentTrack.reset();
+    m_pCurrentTrack = pNewTrack;
```

Comment:
`pTrack` looks redundant

```suggestion
    if (m_pCurrentTrack != nullptr) {
        disconnect(m_pCurrentTrack.get(), nullptr, this, nullptr);
    }
    m_pCurrentTrack = pNewTrack;
```